src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -1510,51 +1532,42 @@ private String quoteUnixCredentials(String str) {
         return str.replace("'", "'\\''");
     }
 
-    private File createWindowsSshAskpass(SSHUserPrivateKey sshUser) throws IOException {
-        File ssh = File.createTempFile("pass", ".bat");
-        try (PrintWriter w = new PrintWriter(ssh)) {
-            w.println("echo \"" + quoteWindowsCredentials(Secret.toString(sshUser.getPassphrase())) + "\"");
-            w.flush();
-        }
-        ssh.setExecutable(true);
+    private FilePath createWindowsSshAskpass(SSHUserPrivateKey sshUser) throws IOException, InterruptedException {
+        FilePath ssh = workspace.createTempFile("pass", ".bat");
+        ssh.write("echo \"" + Secret.toString(sshUser.getPassphrase()) + "\"", null);
+        ssh.chmod(0700);
```

Comment:
If (as @jtnord says) workspace.createTempFile() creates the file in a directory which is not visible from the workspace link in the Jenkins job, then that makes your change much better than the previous implementation.  The previous implementation wrote a plaintext secret to java.io.tmpdir, while your change writes a plaintext secret to a file into that directory hidden inside the Jenkins workspace.

Both the old and new implementation remove the plaintext secret file in the finally block.